client/src/main/java/io/pravega/client/stream/EventWriterConfig.java

```diff
@@ -58,6 +58,14 @@
      * @return Backoff multiplier used in the retry logic of the writer.
      */
     private final int backoffMultiple;
+    
+    /**
+     * The number of milliseconds to wait while attempting to establish a connection before abandoning the attempt and retrying.
+     * 
+     * @param connectionTimeoutMillis The timeout in milliseconds
+     * @return The timeout in milliseconds
+     */
+    private final int connectionTimeoutMillis;
```

Comment:
Nit: Should we rename this field to prevent users from getting confused with Tcp Socket CONNECTION_TIMEOUT ? https://github.com/pravega/pravega/blob/f5ccb4c62087d37fe94fd116aa3c1e50b79d71b6/client/src/main/java/io/pravega/client/connection/impl/TcpClientConnection.java#L68